toolsrc/src/vcpkg/base/downloads.cpp

```diff
@@ -11,6 +14,141 @@
+                                                             : WINHTTP_ACCESS_TYPE_DEFAULT_PROXY,
+                                 WINHTTP_NO_PROXY_NAME,
+                                 WINHTTP_NO_PROXY_BYPASS,
+                                 0);
+            Checks::check_exit(VCPKG_LINE_INFO, h, "WinHttpOpen() failed: %d", GetLastError());
+            m_hSession.reset(h);
+
+            // If the environment variable HTTPS_PROXY is set
+            // use that variable as proxy. This situation might exist when user is in a company network
+            // with restricted network/proxy settings
+            auto maybe_https_proxy_env = System::get_environment_variable("HTTPS_PROXY");
+            if (auto p_https_proxy = maybe_https_proxy_env.get())
+            {
+                std::wstring env_proxy_settings = Strings::to_utf16(*p_https_proxy);
+                WINHTTP_PROXY_INFO proxy;
+                proxy.dwAccessType = WINHTTP_ACCESS_TYPE_NAMED_PROXY;
+                proxy.lpszProxy = env_proxy_settings.data();
+                proxy.lpszProxyBypass = nullptr;
+
+                WinHttpSetOption(m_hSession.get(), WINHTTP_OPTION_PROXY, &proxy, sizeof(proxy));
+            }
+            // Win7 IE Proxy fallback
+            else if (IsWindows7OrGreater() && !IsWindows8Point1OrGreater())
+            {
+                // First check if any proxy has been found automatically
+                WINHTTP_PROXY_INFO proxyInfo;
+                DWORD proxyInfoSize = sizeof(WINHTTP_PROXY_INFO);
+                auto noProxyFound =
+                    !WinHttpQueryOption(m_hSession.get(), WINHTTP_OPTION_PROXY, &proxyInfo, &proxyInfoSize) ||
+                    proxyInfo.dwAccessType == WINHTTP_ACCESS_TYPE_NO_PROXY;
+
+                // If no proxy was found automatically, use IE's proxy settings, if any
+                if (noProxyFound)
+                {
+                    WINHTTP_CURRENT_USER_IE_PROXY_CONFIG ieProxy;
+                    if (WinHttpGetIEProxyConfigForCurrentUser(&ieProxy) && ieProxy.lpszProxy != nullptr)
+                    {
+                        WINHTTP_PROXY_INFO proxy;
+                        proxy.dwAccessType = WINHTTP_ACCESS_TYPE_NAMED_PROXY;
+                        proxy.lpszProxy = ieProxy.lpszProxy;
+                        proxy.lpszProxyBypass = ieProxy.lpszProxyBypass;
+                        WinHttpSetOption(m_hSession.get(), WINHTTP_OPTION_PROXY, &proxy, sizeof(proxy));
+                        GlobalFree(ieProxy.lpszProxy);
+                        GlobalFree(ieProxy.lpszProxyBypass);
+                        GlobalFree(ieProxy.lpszAutoConfigUrl);
+                    }
+                }
+            }
+
+            // Use Windows 10 defaults on Windows 7
+            DWORD secure_protocols(WINHTTP_FLAG_SECURE_PROTOCOL_SSL3 | WINHTTP_FLAG_SECURE_PROTOCOL_TLS1 |
+                                   WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_1 | WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2);
```

Comment:
```suggestion
            DWORD secure_protocols(WINHTTP_FLAG_SECURE_PROTOCOL_TLS1 |
                                   WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_1 | WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2);
```

We probably shouldn't support SSL3; TLS 1.1 and TLS 1.2 are vulnerable to downgrade attacks that leave you with SSL3 (and its known vulnerabilities like CRIME and POODLE). Note that major browsers dropped SSL3 fallback circa 2015.